worlds/earthbound/__init__.py

```diff
@@ -0,0 +1,605 @@
+
+    def pre_fill(self) -> None:
+        prefill_locations = []
+        prefill_items = []
+
+        if not self.options.character_shuffle:
+            main_characters = ["Ness", "Paula", "Jeff", "Poo"]
+            for character in main_characters:
+                if character != self.starting_character:
+                    prefill_items.append(self.create_item(character))
+
+            prefill_items.extend([
+                self.create_item("Flying Man"),
+                self.create_item("Teddy Bear"),
+                self.create_item("Super Plush Bear")
+            ])
+
+            prefill_locations.extend([
+                self.multiworld.get_location("Happy-Happy Village - Prisoner", self.player),
+                self.multiworld.get_location("Threed - Zombie Prisoner", self.player),
+                self.multiworld.get_location("Snow Wood - Bedroom", self.player),
+                self.multiworld.get_location("Monotoli Building - Monotoli Character", self.player),
+                self.multiworld.get_location("Dalaam - Throne Character", self.player),
+                self.multiworld.get_location("Deep Darkness - Barf Character", self.player),
+            ])
+            self.random.shuffle(prefill_locations)
+            add_item_rule(self.multiworld.get_location("Happy-Happy Village - Prisoner", self.player), lambda item: item.name in self.item_name_groups["Characters"])
+            add_item_rule(self.multiworld.get_location("Threed - Zombie Prisoner", self.player), lambda item: item.name in self.item_name_groups["Characters"])
+            add_item_rule(self.multiworld.get_location("Snow Wood - Bedroom", self.player), lambda item: item.name in self.item_name_groups["Characters"])
+            add_item_rule(self.multiworld.get_location("Monotoli Building - Monotoli Character", self.player), lambda item: item.name in self.item_name_groups["Characters"])
+            add_item_rule(self.multiworld.get_location("Dalaam - Throne Character", self.player), lambda item: item.name in self.item_name_groups["Characters"])
+            add_item_rule(self.multiworld.get_location("Deep Darkness - Barf Character", self.player), lambda item: item.name in self.item_name_groups["Characters"])
+
+        fill_restrictive(self.multiworld, self.multiworld.get_all_state(False), prefill_locations, prefill_items, True, True)
+        setup_hints(self)
+
+    def get_pre_fill_items(self) -> list[Item]:
+        characters = ["Ness", "Paula", "Jeff", "Poo"]
+        prefill_items = []
+        for character in characters:
+            if character != self.starting_character:
+                prefill_items.append(self.create_item(f"{character}"))     
+        return prefill_items
+
+    @classmethod
+    def stage_generate_output(cls, multiworld: MultiWorld, output_directory: str) -> None:
+        multiworld.eb_spheres = list(multiworld.get_spheres())
```

Comment:
I consulted the hive mind :tm: and it seems most of us would greatly prefer this to be `cls.eb_spheres` instead of `multiworld.eb_spheres`.

You should then be able to access it via `world.eb_spheres` because of the way attribute resolution on Python classes works (`world.eb_spheres` first checks if it's available on the instance, and if not, it falls back to the class)

Reply:
The hive mind has consulted again and we have decided: Nevermind

Although I think it'd be nice if you gave it a more complicated/longer name so it's less likely to collide with other worlds. If anyone does an implementation for the game "Eternal Battle" there is a non zero chance they will also do `multiworld.eb_spheres` and then suddenly get weird bugs

`multiworld.earthbound_spheres` should be fine, if you wanna be extra safe you could do `earthbound_pinkswitch_spheres`, idk, whatever I'm not gonna be picky about this dw